install/kubernetes/helm/istio/templates/crds.yaml

```diff
@@ -633,7 +633,7 @@ spec:
   scope: Namespaced
   version: v1alpha2
 ---
-
+{{- if .Values.global.proxy.envoyStatsd.enabled }}
```

Comment:
Not clear why there is no condition check for other CRDs in this file.

@linsun any comments for this? I found that you comment out those conditions.

Reply:
Please don't add conditions on CRDs - all CRDs should be deployed ( to support future a-la-carte and upgrades).

Even if a component is disabled - it doesn't mean the configs should be rejected and deployments using the configs should fail. Maybe an alternative implementation is used, or maybe the component will be added later. 

Plus updating CRDs when you change options is a huge pain.